### DISCUSSION TITLE
How to indicate form isDirty, when values are populated post initial render

### DISCUSSION THREAD

**Discussion by GalPayEm** [react-hook-form/react-hook-form#9321] [Q&A] [answered]
Hi,

This is more of a conceptual than "code specific" issue, so bare with the lack of code... 
A controlled form is created, an empty one. after that, an API is called returning values that should be displayed as the "initial values" of said form. Currently, we used `resetField` and reset each field individually once the API returns, and its `defaultValue` becomes the one from the API. This achieved field level `isDirty` indication which works. Unfortunately, the `formState.isDirty` clearly states it doesn't necessarily represent the value of the fields. I want to identify if a form was changed or not once the resetFields happened.

Is there anywhere in RHF a way to know that my fields arent dirty, without looping through them all? is there a better way to reset? I have seen a previous post regarding using `.reset{ defaultValues: defaultValues}`... Is my issue the fact that im using multiple `resetFields` rather than the full reset?

**Comment by Moshyfawn** [comment, marked as answer]
[The `resetField` method](https://react-hook-form.com/api/useform/resetfield) manages the field's state.

You want to use [the `reset` method](https://react-hook-form.com/api/useform/reset) to reset the whole form / manipulate the form's state.

If you want to mark your form as dirty after the form's values reset _(`isDirty === true`)_, you want to provide the `keepDefaultValues: true` flag to [the `reset` method](https://react-hook-form.com/api/useform/reset)'s options.
```js
useEffect(() => {
  if (data) {
    reset(data, { keepDefaultValues: true })
  }
}, [data, reset])
```

**Reply by GalPayEm** [reply]
I want the other way around, when the forms values reset (from API) for` isDirty === false`, I guess `keepValues` would do the trick? apologies for the fairly noobish question, coding the solution isn't my problem, I am trying to understand whats one of the "right ways" of doing this in RHF :)

**Reply by Moshyfawn** [reply]
Well, you want [the `reset` method](https://react-hook-form.com/api/useform/reset) for sure, since you're after manipulation the form's state and not the fields'.

If you just call `reset` with data without any options, `isDirty` is gonna be set to `false`, since the passed data is gonna be the new default values, hence the form `isDirty === false`
